direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorage.java

```diff
@@ -770,7 +846,7 @@ public RandomOffset fetchOffset(Listing type, String key) {
 
     @Override
     public List<Partition> getPartitions(long startStamp, long endStamp) {
-      return Collections.singletonList(PARTITION);
+      return Collections.singletonList(SINGLE_PARTITION);
```

Comment:
I'm reading this very quickly, but - is this correct even when I pass NUM_PARTITIONS > 1?

Reply:
I didn't focus on RandomAccessReader at all, I think it can throw an exception for now when numPartitions > 1, anyway it should be fairly straightforward to add support in the future